packages/beacon-node/src/network/peers/peerManager.ts

```diff
@@ -733,26 +734,47 @@ export class PeerManager {
       agentClient: null,
       encodingPreference: null,
     };
-    this.connectedPeers.set(remotePeer.toString(), peerData);
+    this.connectedPeers.set(remotePeerIdStr, peerData);
 
     if (direction === "outbound") {
       // this.pingAndStatusTimeouts();
       void this.requestPing(remotePeer);
       void this.requestStatus(remotePeer, this.statusCache.get());
     }
 
-    this.libp2p.services.identify
-      .identify(evt.detail)
-      .then((result) => {
-        const agentVersion = result.agentVersion;
-        if (agentVersion) {
-          peerData.agentVersion = agentVersion;
-          peerData.agentClient = getKnownClientFromAgentVersion(agentVersion);
-        }
-      })
-      .catch((err) => {
-        this.logger.debug("Error setting agentVersion for the peer", {peerId: peerData.peerId.toString()}, err);
```

Comment:
can you just check the connection status again inside the catch?
eg:
```ts
.catch((err) => {
  if (conn.status === "closing" || conn.status === "closed") {
     ...
   } else {
     ...
   }
});

Reply:
Don't get your comment. 

If you mean to check status in the refactored code then we don't need it as the abort error will only happen if the connection is not open. 

If you mean to just check in the old code and not refactor, then this will not will fix the issue of not initiating the identify request for a closing connection.

Reply:
This PR doesn't prevent initiating the identity request in some additional way, it only prevents resolving the promise.

imo the problem is not that we initiate an identity request that fails, the problem is the noisy log that doesn't tell us why the error is happening.

Reply:
Yes, my approach was not initiate the request at all or abort early. But anyhow I will update the PR to only conditionally log that error.